pcsx2/Frontend/INISettingsInterface.cpp

```diff
@@ -0,0 +1,242 @@
+	if (!parsed_value.has_value())
+		return false;
+
+	*value = parsed_value.value();
+	return true;
+}
+
+bool INISettingsInterface::GetDoubleValue(const char* section, const char* key, double* value) const
+{
+	const char* str_value = m_ini.GetValue(section, key);
+	if (!str_value)
+		return false;
+
+	std::optional<double> parsed_value = StringUtil::FromChars<double>(str_value);
+	if (!parsed_value.has_value())
+		return false;
+
+	*value = parsed_value.value();
+	return true;
+}
+
+bool INISettingsInterface::GetBoolValue(const char* section, const char* key, bool* value) const
+{
+	const char* str_value = m_ini.GetValue(section, key);
+	if (!str_value)
+		return false;
+
+	std::optional<bool> parsed_value = StringUtil::FromChars<bool>(str_value);
+	if (!parsed_value.has_value())
+		return false;
+
+	*value = parsed_value.value();
+	return true;
+}
+
+bool INISettingsInterface::GetStringValue(const char* section, const char* key, std::string* value) const
+{
+	const char* str_value = m_ini.GetValue(section, key);
+	if (!str_value)
+		return false;
+
+	value->assign(str_value);
+	return true;
+}
+
+void INISettingsInterface::SetIntValue(const char* section, const char* key, int value)
+{
+	m_dirty = true;
+	m_ini.SetLongValue(section, key, static_cast<long>(value), nullptr, false, true);
+}
+
+void INISettingsInterface::SetUIntValue(const char* section, const char* key, uint value)
+{
+	m_dirty = true;
+	m_ini.SetLongValue(section, key, static_cast<long>(value), nullptr, false, true);
+}
+
+void INISettingsInterface::SetFloatValue(const char* section, const char* key, float value)
+{
+	m_dirty = true;
+	m_ini.SetDoubleValue(section, key, static_cast<double>(value), nullptr, true);
+}
+
+void INISettingsInterface::SetDoubleValue(const char* section, const char* key, double value)
+{
+	m_dirty = true;
+	m_ini.SetDoubleValue(section, key, value, nullptr, true);
+}
+
+void INISettingsInterface::SetBoolValue(const char* section, const char* key, bool value)
+{
+	m_dirty = true;
+	m_ini.SetBoolValue(section, key, value, nullptr, true);
+}
+
+void INISettingsInterface::SetStringValue(const char* section, const char* key, const char* value)
+{
+	m_dirty = true;
+	m_ini.SetValue(section, key, value, nullptr, true);
+}
+
+void INISettingsInterface::DeleteValue(const char* section, const char* key)
+{
+	m_dirty = true;
+	m_ini.Delete(section, key);
+}
+
+void INISettingsInterface::ClearSection(const char* section)
+{
+	m_dirty = true;
+	m_ini.Delete(section, nullptr);
+	m_ini.SetValue(section, nullptr, nullptr);
+}
+
+std::vector<std::string> INISettingsInterface::GetStringList(const char* section, const char* key)
+{
+	std::list<CSimpleIniA::Entry> entries;
+	if (!m_ini.GetAllValues(section, key, entries))
+		return {};
+
+	std::vector<std::string> ret;
+	ret.reserve(entries.size());
+	std::transform(entries.begin(), entries.end(), std::back_inserter(ret),
+		[](const CSimpleIniA::Entry& it) { return std::string(it.pItem); });
```

Comment:
Any specific reason why something like below won't work? (just curious, no need to change it as in the end it probably doesn't matter)
```
for (const CSimpleIniA::Entry& it : entries)
     ret.push_back(str::string(it.pItem));
```